src/main/java/org/betonquest/betonquest/quest/event/spawn/SpawnMobEvent.java

```diff
@@ -0,0 +1,97 @@
+package org.betonquest.betonquest.quest.event.spawn;
+
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.quest.event.nullable.NullableEvent;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.instruction.variable.VariableNumber;
+import org.betonquest.betonquest.instruction.variable.VariableString;
+import org.betonquest.betonquest.instruction.variable.location.VariableLocation;
+import org.bukkit.Location;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Mob;
+import org.bukkit.persistence.PersistentDataType;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Spawns mobs at given location, with given equipment and drops.
+ */
+public class SpawnMobEvent implements NullableEvent {
+
+    /**
+     * The location to spawn the mob at.
+     */
+    private final VariableLocation variableLocation;
+
+    /**
+     * The type of mob to spawn.
+     */
+    private final EntityType type;
+
+    /**
+     * The equipment and drops of the mob.
+     */
+    private final Equipment equipment;
+
+    /**
+     * The amount of mobs to spawn.
+     */
+    private final VariableNumber amount;
+
+    /**
+     * The name of the mob.
+     */
+    @Nullable
+    private final VariableString name;
+
+    /**
+     * The marked variable.
+     */
+    @Nullable
+    private final VariableString marked;
+
+    /**
+     * Creates a new spawn mob event.
+     *
+     * @param variableLocation the location to spawn the mob at
+     * @param type             the type of mob to spawn
+     * @param equipment        the equipment and drops of the mob
+     * @param amount           the amount of entities to spawn
+     * @param name             the name of the mob
+     * @param marked           the marked variable
+     * @throws InstructionParseException if the entity type is not a mob
+     */
+    public SpawnMobEvent(final VariableLocation variableLocation, final EntityType type, final Equipment equipment,
+                         final VariableNumber amount, @Nullable final VariableString name, @Nullable final VariableString marked) throws InstructionParseException {
+        if (type.getEntityClass() == null || !Mob.class.isAssignableFrom(type.getEntityClass())) {
+            throw new InstructionParseException("The entity type must be a mob");
+        }
```

Comment:
We have this check now twice, one here and one in the factory. I think here is the right place and the factory dont need to check this.